app/controllers/v3/deployments_controller.rb

```diff
@@ -59,10 +62,12 @@ def create
 
   def update
     deployment = DeploymentModel.find(guid: hashed_params[:guid])
-    resource_not_found!(:deployment) unless deployment &&
-      permission_queryer.can_read_from_space?(deployment.app.space.guid, deployment.app.space.organization.guid)
-    unauthorized! unless permission_queryer.can_write_to_active_space?(deployment.app.space.guid)
-    suspended! unless permission_queryer.is_space_active?(deployment.app.space.guid)
+    resource_not_found!(:deployment) unless deployment
+
+    space = deployment.app.space
```

Comment:
> In some places I noticed us doing multiple DB queries in order to get different attributes of the same thing ...

`deployment.app.space.id` and `deployment.app.space.organization_id` would not lead to two DB queries; sequel fetches data from the database into the object (e.g. `deployment.app.space`) when it is accessed for the first time.

But using a variable `space = deployment.app.space` here makes the code definitely more readable. In other places it might not be necessary.

Reply:
Ah right, my mistake. Thanks for that!